Modules/Module_Import/Import/xASL_imp_DCM2NII_Subject_StartConversion.m

```diff
@@ -58,12 +58,27 @@
             %% Start the conversion. Note that the dicom filter is only in effect when a directory is specified as input.
             try
                 % First we try to see if there are deeper layers
-                [fPath, ~, fExt] = xASL_fileparts(scanpath);
-                if isempty(fExt) % run this part only for folders
-                    filepathsNonRecursive = xASL_adm_GetFileList(fPath, x.modules.import.imPar.dcmExtFilter, 'FPList');
-                    filepathsRecursive = xASL_adm_GetFileList(fPath, x.modules.import.imPar.dcmExtFilter, 'FPListRec');
-                    if isempty(filepathsNonRecursive) && ~isempty(filepathsRecursive)
-                        scanpath = fileparts(filepathsRecursive{1});
+                if xASL_exist(scanpath, 'dir') == 7 % run this part only for folders
+                    filepathsNonRecursive = xASL_adm_GetFileList(scanpath, x.modules.import.imPar.dcmExtFilter, 'FPList'); % List files directly in scanpath
+                    filepathsRecursive = xASL_adm_GetFileList(scanpath, x.modules.import.imPar.dcmExtFilter, 'FPListRec'); % List all files including subdirectories
+                    if isempty(filepathsNonRecursive) && ~isempty(filepathsRecursive) 
+						% There are no files in the directory but there are files in the subdirectories
+						for iFile = 1:length(filepathsRecursive)
+							% Remove the filename and keep only the path
+							filepathsRecursive{iFile} = xASL_fileparts(filepathsRecursive{iFile});
+						end
+						% Find the unique paths
+						filepathsRecursiveUnique = unique(filepathsRecursive);
+						if isempty(filepathsRecursiveUnique)
+							% No files found recursively -> error
+							error(['No files found in ' scanpath]);
+						elseif length(filepathsRecursiveUnique) > 1
```

Comment:
What happens if a user has an ASL sequence, where the user has sorted on SeriesDescription, which in e.g. Siemens results in different folders for control label etc. And the user wants dicom2nii to concatenate them in a single NII, that is probably not something that will work right? Just asking